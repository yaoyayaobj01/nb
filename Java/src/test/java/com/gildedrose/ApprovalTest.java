package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

public class ApprovalTest {
//    @Test
//    void testApproval() {
//        Item[] items = new Item[]{new Item("foo", 0, 0)};
//        GildedRose app = new GildedRose(items);
//        app.updateQuality();
//        Approvals.verifyAll("", app.items);
//    }

    @Test
    public void updateQuality() {
        CombinationApprovals.verifyAllCombinations(
            this::callUpdateQuality,
            new String[]{"a common item", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"},
            new Integer[]{0, -1, 20}, //sellIn
            new Integer[]{0, 1, 80, 49, 50}  //quality
        );
    }

    private String callUpdateQuality(String name, int sellIn, int quality) {
        Item[] items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose gildedRose = new GildedRose(items);
        gildedRose.updateQuality();
        return gildedRose.items[0].toString().replace(" ", "");
    }
}
