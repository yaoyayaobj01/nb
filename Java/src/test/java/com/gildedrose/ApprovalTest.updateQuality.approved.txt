[a common item, 0, 0] => acommonitem,-1,0 
[a common item, 0, 1] => acommonitem,-1,0 
[a common item, 0, 80] => acommonitem,-1,78 
[a common item, 0, 49] => acommonitem,-1,47 
[a common item, 0, 50] => acommonitem,-1,48 
[a common item, -1, 0] => acommonitem,-2,0 
[a common item, -1, 1] => acommonitem,-2,0 
[a common item, -1, 80] => acommonitem,-2,78 
[a common item, -1, 49] => acommonitem,-2,47 
[a common item, -1, 50] => acommonitem,-2,48 
[a common item, 20, 0] => acommonitem,19,0 
[a common item, 20, 1] => acommonitem,19,0 
[a common item, 20, 80] => acommonitem,19,79 
[a common item, 20, 49] => acommonitem,19,48 
[a common item, 20, 50] => acommonitem,19,49 
[Aged Brie, 0, 0] => AgedBrie,-1,2 
[Aged Brie, 0, 1] => AgedBrie,-1,3 
[Aged Brie, 0, 80] => AgedBrie,-1,80 
[Aged Brie, 0, 49] => AgedBrie,-1,50 
[Aged Brie, 0, 50] => AgedBrie,-1,50 
[Aged Brie, -1, 0] => AgedBrie,-2,2 
[Aged Brie, -1, 1] => AgedBrie,-2,3 
[Aged Brie, -1, 80] => AgedBrie,-2,80 
[Aged Brie, -1, 49] => AgedBrie,-2,50 
[Aged Brie, -1, 50] => AgedBrie,-2,50 
[Aged Brie, 20, 0] => AgedBrie,19,1 
[Aged Brie, 20, 1] => AgedBrie,19,2 
[Aged Brie, 20, 80] => AgedBrie,19,80 
[Aged Brie, 20, 49] => AgedBrie,19,50 
[Aged Brie, 20, 50] => AgedBrie,19,50 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => BackstagepassestoaTAFKAL80ETCconcert,-1,0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => BackstagepassestoaTAFKAL80ETCconcert,-1,0 
[Backstage passes to a TAFKAL80ETC concert, 0, 80] => BackstagepassestoaTAFKAL80ETCconcert,-1,0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => BackstagepassestoaTAFKAL80ETCconcert,-1,0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => BackstagepassestoaTAFKAL80ETCconcert,-1,0 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => BackstagepassestoaTAFKAL80ETCconcert,-2,0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => BackstagepassestoaTAFKAL80ETCconcert,-2,0 
[Backstage passes to a TAFKAL80ETC concert, -1, 80] => BackstagepassestoaTAFKAL80ETCconcert,-2,0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => BackstagepassestoaTAFKAL80ETCconcert,-2,0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => BackstagepassestoaTAFKAL80ETCconcert,-2,0 
[Backstage passes to a TAFKAL80ETC concert, 20, 0] => BackstagepassestoaTAFKAL80ETCconcert,19,1 
[Backstage passes to a TAFKAL80ETC concert, 20, 1] => BackstagepassestoaTAFKAL80ETCconcert,19,2 
[Backstage passes to a TAFKAL80ETC concert, 20, 80] => BackstagepassestoaTAFKAL80ETCconcert,19,80 
[Backstage passes to a TAFKAL80ETC concert, 20, 49] => BackstagepassestoaTAFKAL80ETCconcert,19,50 
[Backstage passes to a TAFKAL80ETC concert, 20, 50] => BackstagepassestoaTAFKAL80ETCconcert,19,50 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras,HandofRagnaros,0,0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras,HandofRagnaros,0,1 
[Sulfuras, Hand of Ragnaros, 0, 80] => Sulfuras,HandofRagnaros,0,80 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras,HandofRagnaros,0,49 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras,HandofRagnaros,0,50 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras,HandofRagnaros,-1,0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras,HandofRagnaros,-1,1 
[Sulfuras, Hand of Ragnaros, -1, 80] => Sulfuras,HandofRagnaros,-1,80 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras,HandofRagnaros,-1,49 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras,HandofRagnaros,-1,50 
[Sulfuras, Hand of Ragnaros, 20, 0] => Sulfuras,HandofRagnaros,20,0 
[Sulfuras, Hand of Ragnaros, 20, 1] => Sulfuras,HandofRagnaros,20,1 
[Sulfuras, Hand of Ragnaros, 20, 80] => Sulfuras,HandofRagnaros,20,80 
[Sulfuras, Hand of Ragnaros, 20, 49] => Sulfuras,HandofRagnaros,20,49 
[Sulfuras, Hand of Ragnaros, 20, 50] => Sulfuras,HandofRagnaros,20,50 
